app/stac_api/views.py

```diff
@@ -126,6 +127,14 @@ class CollectionList(generics.GenericAPIView, mixins.CreateModelMixin):
     serializer_class = CollectionSerializer
     queryset = Collection.objects.all()
 
+    def get_serializer(self, *args, **kwargs):
+        serializer_class = self.get_serializer_class()
+        kwargs.setdefault('context', self.get_serializer_context())
+
+        validate_JSON_payload_write_requests(serializer_class, self.request.data)
+
+        return serializer_class(*args, **kwargs)
```

Comment:
This method should be generalized either in a mixin or a new based class. I would use a mixin. Also it would be better to create the serializer instance then pass it to the validator and after validation return the instance. This would be more efficient and might be required for asset management were we might need serializer dynamic field.

Reply:
:+1:

Reply:
Done.